kahuna/public/js/search/results.js

```diff
@@ -408,21 +408,34 @@ results.controller('SearchResultsCtrl', [
             }
         };
 
-        const freeUpdateListener = $rootScope.$on('image-updated', (e, updatedImage) => {
-            var index = ctrl.images.findIndex(i => i.data.id === updatedImage.data.id);
-            if (index !== -1) {
-                ctrl.images[index] = updatedImage;
-            }
-
-            var indexAll = ctrl.imagesAll.findIndex(i => i && i.data.id === updatedImage.data.id);
-            if (indexAll !== -1) {
-                ctrl.imagesAll[indexAll] = updatedImage;
 
-                // TODO: should not be needed here, the results list
-                // should listen to these events and update itself
-                // outside of any controller.
-                results.set(indexAll, updatedImage);
-            }
+      const freeUpdatesListener = $rootScope.$on('images-updated', (e, updatedImages) => {
+        updatedImages.map(updatedImage => {
+          var index = ctrl.images.findIndex(i => i.data.id === updatedImage.data.id);
+          if (index !== -1) {
+            ctrl.images[index] = updatedImage;
+          }
+
+          var indexAll = ctrl.imagesAll.findIndex(i => i && i.data.id === updatedImage.data.id);
+          if (indexAll !== -1) {
+            ctrl.imagesAll[indexAll] = updatedImage;
+
+            // TODO: should not be needed here, the results list
```

Comment:
Is this commented section still required?

Reply:
yes, but not the `results.set` call and it's a few lines too high now (fixed)

Reply:
👍